src/Processors/Formats/Impl/PrettyBlockOutputFormat.h

```diff
@@ -50,6 +53,38 @@ class PrettyBlockOutputFormat : public IOutputFormat
     void writeValueWithPadding(
         const IColumn & column, const ISerialization & serialization, size_t row_num,
         size_t value_width, size_t pad_to_width, bool align_right);
+
+private:
+    bool mono_block;
+    /// For mono_block == true only
+    Chunk mono_chunk;
 };
 
+template <class OutputFormat>
+void registerPrettyFormatWithNoEscapesAndMonoBlock(FormatFactory & factory, const String & base_name)
+{
+    for (bool no_escapes : {true, false})
+    {
+        String name = no_escapes ? base_name + "NoEscapes" : base_name;
+        for (bool mono_block : {true, false})
+        {
+            factory.registerOutputFormat(mono_block ? name + "MonoBlock" : name, [no_escapes, mono_block](
+                WriteBuffer & buf,
+                const Block & sample,
+                const RowOutputFormatParams &,
+                const FormatSettings & format_settings)
+            {
+                if (no_escapes)
+                {
+                    FormatSettings changed_settings = format_settings;
+                    changed_settings.pretty.color = false;
+                    return std::make_shared<OutputFormat>(buf, sample, changed_settings, mono_block);
+                }
+                return std::make_shared<OutputFormat>(buf, sample, format_settings, mono_block);
+            });
+        }
+        factory.markOutputFormatSupportsParallelFormatting(name);
+    }
```

Comment:
I'd create lambda and call it four times, to reduce code indentation level and probably make intent more explicit. But also up to you, haven't check if snippet works
```suggestion
    auto creator = [&](FormatFactory & factory, const String & name, bool no_escapes, bool mono_block)
    {
        factory.registerOutputFormat(name, [no_escapes, mono_block](
            WriteBuffer & buf,
            const Block & sample,
            const RowOutputFormatParams &,
            const FormatSettings & format_settings)
        {
            if (no_escapes)
            {
                FormatSettings changed_settings = format_settings;
                changed_settings.pretty.color = false;
                return std::make_shared<OutputFormat>(buf, sample, changed_settings, mono_block);
            }
            return std::make_shared<OutputFormat>(buf, sample, format_settings, mono_block);
        });
        if (!mono_block)
            factory.markOutputFormatSupportsParallelFormatting(name);
    };
    creator(factory, base_name, false, false);
    creator(factory, base_name + "NoEscapes", true, false);
    creator(factory, base_name + "MonoBlock", false, true);
    creator(factory, base_name + "NoEscapesMonoBlock", true, true);
```